src/nex_types.rs

```diff
@@ -0,0 +1,446 @@
+        }
+
+        if let Some(stream) = &self.stream {
+            fields.push(stream.to_string());
+        }
+
+        if let Some(sid) = &self.sid {
+            fields.push(sid.to_string());
+        }
+
+        if let Some(cid) = &self.cid {
+            fields.push(cid.to_string());
+        }
+
+        if let Some(pid) = &self.pid {
+            fields.push(pid.to_string());
+        }
+
+        if let Some(transport_type) = &self.transport_type {
+            fields.push(transport_type.to_string());
+        }
+
+        if let Some(rvcid) = &self.rvcid {
+            fields.push(rvcid.to_string());
+        }
+
+        if let Some(natm) = &self.natm {
+            fields.push(natm.to_string());
+        }
+
+        if let Some(natf) = &self.natf {
+            fields.push(natf.to_string());
+        }
+
+        if let Some(upnp) = &self.upnp {
+            fields.push(upnp.to_string());
+        }
+
+        if let Some(pmp) = &self.pmp {
+            fields.push(pmp.to_string());
+        }
+
+        if let Some(probe_init) = &self.probe_init {
+            fields.push(probe_init.to_string());
+        }
+
+        if let Some(prid) = &self.prid {
+            fields.push(prid.to_string());
+        }
+
+        let fields_string = fields.join(";");
+        let scheme = if let Some(scheme) = &self.scheme {
+            scheme.to_string()
+        } else {
+            "".to_string()
+        };
+
+        format!("{}:/{}", scheme, fields_string)
+    }
+}
+
+struct Result {
+    result_code: u32
+}
+
+impl Result {
+    pub fn new_from_code(result_code: u32) -> Self {
+        Self {
+            result_code
+        }
+    }
+
+    pub fn extract_from_stream<T: Reader>(&mut self, stream: &mut StreamIn<T>) -> core::result::Result<(), &'static str> {
+        self.result_code = stream
+            .read_stream_le::<u32>()
+            .map_err(|_| "Result code could not be read")?
+            .try_into()
+            .map_err(|_|"Result code size does not fit into u32")?;
+
+        Ok(())
+    }
+
+    pub fn bytes(&self, mut stream: StreamOut) -> Vec<u8> {
+        stream.checked_write_stream_le(&self.result_code);
+        stream.into()
+    }
+}
+
+struct ResultRange {
+    offset: u32,
+    length: u32
+}
+
+impl ResultRange {
+    pub fn new() -> Self {
+        Self {
+            offset: 0,
+            length: 0
+        }
+    }
+}
+
+impl StructureInterface for ResultRange {
+    fn extract_from_stream<T: Reader>(&mut self, stream: &mut StreamIn<T>) -> std::result::Result<(), &'static str> {
+        self.offset = stream
+            .read_stream_le::<u32>()
+            .map_err(|_| "Offset could not be read")?
+            .try_into()
+            .map_err(|_|"Offset size does not fit into u32")?;
+
+        self.length = stream
+            .read_stream_le::<u32>()
+            .map_err(|_| "Length could not be read")?
+            .try_into()
+            .map_err(|_|"Length size does not fit into u32")?;
```

Comment:
Same as above, we can remove the `try_into`s and use type inference:
```suggestion
        self.offset = stream
            .read_stream_le()
            .map_err(|_| "Offset could not be read")?;

        self.length = stream
            .read_stream_le()
            .map_err(|_| "Length could not be read")?;
```